src/main/java/com/bobocode/hoverla/bring/context/BeanValidator.java

```diff
@@ -1,18 +1,125 @@
+            throw new NoUniqueBeanException("Context contains beans with duplicate names: " + duplicateNames.keySet());
+        }
+    }
+
+    private void beanNameValidation(BeanDefinition beanDefinition) {
+        log.trace("Checking name for bean definition: {} - {}", beanDefinition.name(), beanDefinition.type());
+        String name = beanDefinition.name();
+        if (beanDefinition.name() == null || name.isEmpty() || name.contains(" ")) {
+            throw new IncorrectBeanNameException(String.format("Bean name for %s class is null, empty or contains spaces",
+                    beanDefinition.type().getSimpleName()));
+        }
+    }
+
+    private void dependenciesValidation(BeanDefinition beanDefinition,
+                                        List<BeanDefinition> allBeanDefinitions) {
+        log.trace("Checking dependencies for bean definition: {} - {}", beanDefinition.name(),
+                beanDefinition.type());
+        if (beanDefinition.dependencies().isEmpty()) {
+            return;
+        }
+        var dependencies = beanDefinition.dependencies();
+
+        for (Map.Entry<String, Class<?>> dependency : dependencies.entrySet()) {
+            log.trace("***Checking dependency {} - {}", dependency.getKey(), dependency.getValue());
+            var allBeansByDependencyName = findAllBeanByName(dependency.getKey(), allBeanDefinitions);
+            if (allBeansByDependencyName.size() > 1) {
+                throw new NoUniqueBeanException("Context contains beans with duplicate names: " + dependency.getKey());
+            }
+            if (allBeansByDependencyName.size() == 1) {
+                validateDependencyTypeAndFoundedBeanType(dependency, allBeansByDependencyName);
+                checkCircularDependency(beanDefinition, allBeansByDependencyName);
+            } else {
+                checkBeansByType(dependency.getValue(), allBeanDefinitions);
+            }
+        }
+    }
+
+    private List<BeanDefinition> findAllBeanByName(String source, List<BeanDefinition> allBeanDefinitions) {
+        return allBeanDefinitions.stream()
+                .filter(b -> source.equals(b.name()))
+                .toList();
+    }
+
+    private void validateDependencyTypeAndFoundedBeanType(Map.Entry<String, Class<?>> dependency,
+                                                          List<BeanDefinition> allBeansByName) {
+        Class<?> foundBeanType = allBeansByName.get(0).type();
+        if (!foundBeanType.equals(dependency.getValue())) {
+            String exceptionMessage = "Types are not coincide for required dependency - %s and founded bean - %s";
+            throw new IncorrectBeanTypeException(String.format(exceptionMessage, dependency.getValue(), foundBeanType));
+        }
+    }
+
+    private void checkCircularDependency(BeanDefinition sourceBeanDefinition, List<BeanDefinition> allBeanByName) {
+        if (allBeanByName.get(0).dependencies().containsKey(sourceBeanDefinition.name())) {
```

Comment:
Why do we actually pass the whole list here then, while we can pass the one element

Reply:
done